mautic/mautic
# Allow HTML in custom email token sent via API

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [n]
| New feature/enhancement? (use the a.x branch)      | [enhancement]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

Use case:

A user has an e-shop and wants to send an abandoned cart email. Mautic would need to know about the abandoned cart in order to replace the token correctly. However, the e-shop may send the Mautic email via API and provide the abandoned cart content as a custom token.

Sending custom tokens with email send is already possible:

https://developer.mautic.org/#send-email-to-contact

However, if the token contains some HTML like `custom <b>value</b>` then it will be sent in the email body like `custom &#60;b&#62;value&#60;/b&#62;`. This PR changes the sanitization of the custom token values to allowing HTML characters.

#### Steps to test this PR:

The API call is tested in the modified functional test so I don't think it needs any special user testing, but if anyone wants to go ahead:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Use https://developer.mautic.org/#send-email-to-contact to send an email with custom tokens. Check the PHPUNIT test if you'll need some help setting it up.
